cci_nepal/analysis/dummy_data/dummy_data_modeling/model_analysis.py

```diff
@@ -0,0 +1,295 @@
+import pandas as pd
+import numpy as np
+from numpy import absolute
+from numpy import mean
+from numpy import std
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import cross_val_score
+from sklearn.model_selection import RepeatedKFold
+from sklearn.feature_selection import SequentialFeatureSelector
+from matplotlib import pyplot
+from operator import itemgetter
+
+import logging
+import cci_nepal
+from cci_nepal.getters.dummy_data import get_dummy_data as gdd
+
+
+def label_transformer(d):
+    if d < 1.5:
+        return "Unnecessary"
+    elif d < 2.5:
+        return "Desirable"
+    else:
+        return "Essential"
+
+
+def numeric_score_transformer(d):
+    if d < 1.5:
+        return 1
+    elif d < 2.5:
+        return 2
+    else:
+        return 3
+
+
+project_dir = cci_nepal.PROJECT_DIR
+logging.info(project_dir)
+
+
+df = gdd.read_csv_file(f"{project_dir}/inputs/data/Dummy_Data_Full_Survey_Features.csv")
+df.head()
+
+
+df.columns
+
+# Mapping categories to numbers #
+
+mapping = {"Essential": 3, "Desirable": 2, "Unnecessary": 1}
+df_dummy = df.applymap(lambda s: mapping.get(s) if s in mapping else s)
+
+
+df_dummy.head(5)
+
+# This part is just to test if all 3 labels are predicted well in the model ###
+
+# Since most of the predictions were otherwise averaged around 2 ###
+
+
+def replace_with_number(df, columns, values):
+    for one_column, one_value in zip(columns, values):
+        df[one_column] = one_value
+
+    return df
+
+
+replace_with_number(df_dummy, ["Blanket", "Printed Cloth", "Hand Sanitizer"], [3, 1, 3])
+
+
+# Columns to categorize for Linear Regression modeling #
+
+num_cols = df_dummy._get_numeric_data().columns
+
+columns_to_categorize = list(set(df_dummy.columns) - set(num_cols))
+
+df_dummy[columns_to_categorize] = df_dummy[columns_to_categorize].astype("category")
+
```

Comment:
Some things we need to consider about the features:

1. Many of the features will be correlated with each other, it would be to understand this better in the feature selection phase
2. Although we don't have 'no of household members' as a feature, its sortof captured and held within the gender and age breakdown columns. Having it this way means we might give weight to a particular gender / age feature when its actually about the hh size. For example, there may be a household with 10 people in it 8 of those are males 18-29. This feature may then show as an important feature in predicting certain items, or may even skew the results when its actually about the size of the hh. A solution here could be to create a hh size feature and then tranform the age/gender features to %. So its 80% male 18:29 rather than 8.
3. The same as the above goes for the ethnicity and disabilty columns. Also will mean they will all be correlated as a result (which will affect the model).
4. Do we need to standardise and scale some of the features so they are not skewing the results?
5. I think we should think about how we want to transform the age/gender column now to reduce the features here. This may need to be informed by RC feedback also. One thought is, for under 5s would we need the results separated by gender?
6. The variance of the features is another thing we need to look at when we have the actual data